src/libraries/System.Private.Uri/tests/FunctionalTests/UriTests.cs

```diff
@@ -729,6 +729,41 @@ public static void Uri_CombineUsesNewUriString()
             Assert.Equal(Combined, new Uri(baseUri, RelativeUriString).AbsoluteUri);
         }
 
+        [Theory]
+        [InlineData("http://bar/Testue/testImage.jpg")]
+        // Tests that internal Uri info were properly applied during a Combine operation when URI contains non-ascii character.
+        [InlineData("http://bar/Test�/testImage.jpg")]
+        public static void Uri_CombineWithAbsoluteHttpUriResultInAbsoluteHttpSchema(string fileUri)
+        {
+            string baseUriString = "combine-scheme://foo";
+
+            var baseUri = new Uri(baseUriString, UriKind.Absolute);
+            var uri = new Uri(fileUri);
+            var resultUri = new Uri(baseUri, uri);
+            string resultUriString = resultUri.ToString();
+
+            Assert.DoesNotContain(baseUriString, resultUriString);
+            Assert.Contains("http://", resultUriString);
+            Assert.Equal(fileUri, resultUriString);
+        }
+
+        [Theory]
+        [InlineData(@"\\nas\Testue\testImage.jpg")]
+        // Tests that internal Uri info were properly applied during a Combine operation when URI contains non-ascii character.
+        [InlineData(@"\\nas\Test�\testImage.jpg")]
+        public static void Uri_CombineWithAbsoluteFilePathResultInAbsoluteFileSchema(string fileUri)
```

Comment:
This appears to be the same as the test above, I think we can just expand the inline data for that one with these inputs instead.

Reply:
File is somehow specific, as it implicitly add file schema, which was the main reason I have created two tests. But you are right, one test with given expected values in inline data will be better.